src/rules/effects/no-dispatch-in-effects.ts

```diff
@@ -70,3 +89,36 @@ function getNodeToReport(node: MemberExpressionWithinCallExpression) {
     ? node
     : parent
 }
+
+function isLikelyToContainDispatchFalse({
+  arguments: [, callExpressionArgument],
+}: TSESTree.CallExpression) {
+  if (!callExpressionArgument || !isObjectExpression(callExpressionArgument)) {
+    return false
+  }
+
+  let hasDispatchFalse = false
+
+  for (const property of callExpressionArgument.properties) {
+    if (
+      !isProperty(property) ||
+      (property.computed && isIdentifier(property.key))
+    ) {
+      return true
+    }
+
+    if (!hasDispatchFalse && isDispatchFalse(property)) {
+      hasDispatchFalse = true
```

Comment:
Could this also be a short-circuit and just return `true`?
That way, we can get rid of the `hasDispatchFalse` variable

Reply:
I don't like this either, but I'm afraid we can't.. because we need to loop over **all** the elements since we can't say that it has a dispatch `false` only looking for `dispatch` property... what if we find a non-`Property` (e.g. `SpreadElement`) or a `Identifier`[computed] (`{ [dispatchKey]: false }`)?

Reply:
I think we're talking about something different.
What I mean is that if the condition `isDispatchFalse(property)` passes, that we can return `true` and in that way short-circuit the `for` iteration. In other words, once we know it is a `dispatch: false`, we can ignore the rest of the properties.

Reply:
Great! I misunderstood your statement at first. I've adjusted one of the tests to make it clear + used `.some` instead of `for-of`.